### PR TITLE
dlang: Don't try to convert D2 only projects

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**

### CODE REVIEW

**Comment by joseph-wakeling-sociomantic** [comment]
LGTM in principle.  I'm not going to comment on whether the solution actually works, although happy to test locally if it would help .

**Comment by leandro-lucarella-sociomantic** [comment]
Seeing why it's red and working on some tests (which I should have before sending the PR :D).

**Comment by leandro-lucarella-sociomantic** [comment]
Green

File: bin/dlang/make [hunk 1]
```diff
@@ -22,10 +22,13 @@ beaver=$r/bin/beaver
...
 set_dc_dver
-export BEAVER_DOCKER_VARS="${BEAVER_DOCKER_VARS:-} DC DVER"
+export BEAVER_DOCKER_VARS="${BEAVER_DOCKER_VARS:-} DC DVER D2_ONLY"
 
-# First convert code if we are building D2
-if_d 2 beaver make d2conv
+# First convert code if we are building a D2 version of a D1/2 project
+if test "$D2_ONLY" = false -a "$DVER" -eq 2
+then
+    $beaver make d2conv
```
Comment (nemanja-boric-sociomantic) [inline bin/dlang/make:30]: Ah, it's so much easier with the matrix.

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Pull request merged**